### DISCUSSION TITLE
npm run deploy takes forever

### DISCUSSION THREAD

**Discussion by smakhtin** [mozilla/hubs#2848] [Hubs Cloud Questions] [answered]
Hi everyone. I'm trying to deploy custom Hubs client and see "Building Admin Console" for 20 minutes i think. Is this really a correct build time or something wrong with the build in general? I canceled build under Windows and trying on WSL - same stuff.

**Comment by robinkwilson** [comment, marked as answer]
When I have an issue with the admin panel build, the `npm run deploy` command hangs on the "Building Admin Console". Though sometimes with the `npm run deploy` command, I have to run it once or twice until it goes through. 

To test that your admin panel build is working, in your hubs repo root, try:
1. `cd admin`
2. `npm ci`
3. If it succeeds, try the `npm run deploy` command again.
4. If it fails (might need to try once or twice), fix the error, then try again.

Let me know how it goes.

**Reply by smakhtin** [reply]
Thx, I think this should be added to documentation or we should implement an error handling in deployment command.

**Reply by robinkwilson** [reply]
Issue created: https://github.com/mozilla/hubs/issues/2873
Docs updated: https://github.com/MozillaReality/hubs-docs/blob/master/docs/hubs-cloud-custom-clients.md
